ctapipe/reco/shower_processor.py

```diff
@@ -0,0 +1,193 @@
+        self._is_simulation = is_simulation
+        self.reconstruct_energy = reconstruct_energy
+        self.classify = classify
+        self.reconstructor = HillasReconstructor()
+
+    def _reconstruct_shower(
+        self,
+        event,
+        default=DEFAULT_SHOWER_PARAMETERS,
+    ) -> ReconstructedShowerContainer:
+        """Perform shower reconstruction.
+
+        Parameters
+        ----------
+        tel_id: int
+            which telescope is being cleaned
+        image: np.ndarray
+            image to process
+        signal_pixels: np.ndarray[bool]
+            image mask
+        peak_time: np.ndarray
+            peak time image
+        Returns
+        -------
+        ReconstructedShowerContainer:
+            direction in the sky with uncertainty,
+            core position on the ground with uncertainty,
+            h_max with uncertainty,
+            is_valid boolean for successfull reconstruction,
+            average intensity of the intensities used for reconstruction,
+            measure of algorithm success (if fit),
+            list of tel_ids used if stereo, or None if Mono
+        """
+
+        # Read only valid HillasContainers (minimum condition to continue)
+        hillas_dict = {
+            tel_id: dl1.parameters.hillas
+            for tel_id, dl1 in event.dl1.tel.items()
+            if np.isfinite(event.dl1.tel[tel_id].parameters.hillas.intensity)
+        }
+
+        # On top of this check if the shower should be considered based
+        # on the user's configuration
+        shower_criteria = [self.check_shower(hillas_dict[tel_id]) for tel_id in hillas_dict]
+        self.log.debug(
+            "shower_criteria: %s",
+            list(zip(self.check_shower.criteria_names[1:], shower_criteria)),
+        )
+
+        # Reconstruct the shower only if all shower criteria are met
+        if np.count_nonzero(shower_criteria) > 2:
+
+            array_pointing = SkyCoord(
+                az=event.pointing.array_azimuth,
+                alt=event.pointing.array_altitude,
+                frame=AltAz(),
+            )
+
+            telescopes_pointings = {
+                tel_id: SkyCoord(
+                    alt=event.pointing.tel[tel_id].altitude,
+                    az=event.pointing.tel[tel_id].azimuth,
+                    frame=AltAz(),
+                )
+                for tel_id in hillas_dict
+            }
+
+            result = self.reconstructor.predict(hillas_dict,
+                                                self.subarray,
+                                                array_pointing,
+                                                telescopes_pointings)
+
+            return result
+
+        else:
+            return default
+
+    def _reconstruct_energy(self, event: ArrayEventContainer):
+        raise NotImplementedError("TO DO")
+
+    def _estimate_classification(self, event: ArrayEventContainer):
```

Comment:
Be more explicit. e.g. `classify_particle_type`

Reply:
done